CHANGELOG.md

```diff
@@ -40,6 +40,7 @@ FEATURES:
 
 * **New Data Source:** `azurerm_eventgrid_system_topic` ([#13851](https://github.com/hashicorp/terraform-provider-azurerm/issues/13851))
 * **New Data Source:** `azurerm_billing_mpa_account_scope` ([#13723](https://github.com/hashicorp/terraform-provider-azurerm/issues/13723))
+* **New Data Source:** `azurerm_batch_application` ([#14043](https://github.com/hashicorp/terraform-provider-azurerm/issues/14043))
```

Comment:
While the foresight is appreciated here, the changelog entry is only done after the PR has been approved and merged.
```suggestion
```